server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataManager.java

```diff
@@ -1063,12 +1063,20 @@ public DataSegment map(int index, ResultSet r, StatementContext ctx) throws SQLE
     if (segments.isEmpty()) {
       log.info("No segments found in the database!");
     } else {
-      log.info("Polled and found %,d segments in the database", segments.size());
+      log.info("Polled and found %,d segments in the database.", segments.size());
     }
+
+    log.debug("Building datasources snapshot from polled segments.");
+    final long snapshotCreationStartTime = System.currentTimeMillis();
     dataSourcesSnapshot = DataSourcesSnapshot.fromUsedSegments(
         Iterables.filter(segments, Objects::nonNull), // Filter corrupted entries (see above in this method).
         dataSourceProperties
     );
+    final long snapshotCreationEndTime = System.currentTimeMillis();
```

Comment:
Instead of this, initialize a Druid `Stopwatch` at the start of this method. And in every log, print the result of `.millisElapsed()` to get a clear idea of the total time spent after performing each step.

Reply:
Thanks! Done.